### DISCUSSION TITLE
help with query field

### DISCUSSION THREAD

**Discussion by the1gofer** [mdelobelle/metadatamenu#542] [Q&A]
I have the current query that works:

```dataviewjs
dv.list(dv.pages('"TCPA/Notes/Documents"').where(p => dv.func.contains(p.linkedNotes, dv.current().file.name)).file.name)
```

For the life of me, I cannot figure out how to get this to work in a query field. Any suggestions?  I've tried 
```
dv.pages('"TCPA/Notes/Documents"').where(p => dv.func.contains(p.linkedNotes, dv.current().file.name)).file.name
```
with linkedNotes in the "name of related field".  

Can someone offer any guidance?

**Comment by mdelobelle** [comment]
you could try to replace dv.current() with `current`

the reason is that the query isn't executed in the context of the note like when it is embeded in a code block, so we have to tell it what this context is. Metadata Menu does that for us by passing dv.page(...file path of the file where you are using the field...) to the constant `current`. Therefore `current` has exactly the same property as `dv.current()` when called in a code block in the same file.

**Reply by the1gofer** [reply]
thanks.  before I can test this, I've run into another issue.  for readability I have some fields that have spaces in them.  regular dataview allows you to access them through something like `some-field` or `row["some-field"]`.  That doens't work in java script.  do you know how I access those fields? or is the only way to rename them?

**Reply by the1gofer** [reply]
hmmm... so that didn't actually change anything.   what should the target be?
